src/smirnoffio.py

```diff
@@ -560,6 +564,40 @@ def __init__(self,options,tgt_opts,forcefield):
         ## Initialize base class.
         super(Vibration_SMIRNOFF,self).__init__(options,tgt_opts,forcefield)
 
+    def submit_jobs(self, mvals, AGrad=False, AHess=False):
+        # we update the self.pgrads here so it's not overwritten in rtarget.py
+        self.smirnoff_update_pgrads()
+
+    def smirnoff_update_pgrads(self):
```

Comment:
Yes there are duplications, but it's not general for all targets because some targets like `OptGeoTarget` have a layered structure of multiple molecules and multiple engines, so the implementation is different.

Reply:
No problem, I'll resolve this.